src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicPlatformDetectionTests.cs

```diff
@@ -19,5 +20,33 @@ public void UnsupportedPlatforms_ThrowsPlatformNotSupportedException()
             Assert.ThrowsAsync<PlatformNotSupportedException>(async () => await CreateQuicListener());
             Assert.ThrowsAsync<PlatformNotSupportedException>(async () => await CreateQuicConnection(new IPEndPoint(IPAddress.Loopback, 0)));
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows), nameof(PlatformDetection.SupportsTls13))]
+        public void SupportedWindowsPlatforms_IsSupportedIsTrue()
+        {
+            Assert.True(QuicListener.IsSupported);
+            Assert.True(QuicConnection.IsSupported);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsLinux))]
+        public async Task SupportedLinuxPlatforms_IsSupportedIsTrue()
+        {
+            using Process ldconfig = new Process();
+            ldconfig.StartInfo.FileName = "ldconfig";
+            ldconfig.StartInfo.Arguments = "-p";
```

Comment:
This is going to fail on Alpine

docker run -ti mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.14-helix-amd64 bash
```
bash-5.1$ ldconfig -p
Illegal option -p
```

Reply:
I'm not sure what would be better option. Perhaps pinvoke `dlopen()` ?

Reply:
I switched that to `find` and it seems like it caught first problem with alpine:
```
 Discovering: System.Net.Quic.Functional.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Net.Quic.Functional.Tests (found 3 of 114 test cases)
  Starting:    System.Net.Quic.Functional.Tests (parallel test collections = on, max threads = 2)
    System.Net.Quic.Tests.MsQuicPlatformDetectionTests.SupportedWindowsPlatforms_IsSupportedIsTrue [SKIP]
      Condition(s) not met: "IsWindows"
    System.Net.Quic.Tests.MsQuicPlatformDetectionTests.SupportedLinuxPlatforms_IsSupportedIsTrue [FAIL]
      Assert.True() Failure
      Expected: True
      Actual:   False
      Stack Trace:
        /_/src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicPlatformDetectionTests.cs(45,0): at System.Net.Quic.Tests.MsQuicPlatformDetectionTests.SupportedLinuxPlatforms_IsSupportedIsTrue()
        --- End of stack trace from previous location ---
      Output:
        Using unknown
        /usr/lib/libmsquic.so.2
        
  Finished:    System.Net.Quic.Functional.Tests
=== TEST EXECUTION SUMMARY ===
   System.Net.Quic.Functional.Tests  Total: 3, Errors: 0, Failed: 1, Skipped: 1, Time: 0.715s
```
https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-81481-merge-f74ebb134a30429ca3/System.Net.Quic.Functional.Tests/2/console.b17bfdef.log?helixlogtype=result

Reply:
Based on my local test, it fails here:
https://github.com/dotnet/runtime/blob/490b20fcfa6c984d04b8163cb81c9b65029f55c5/src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs#L69-L70

Reply:
Lib is present, but cannot be loaded for some reason. `ldd` fails with:
```
ldd /usr/lib/libmsquic.so.2
        /lib/ld-musl-x86_64.so.1 (0x7fbc0efdf000)
        libcrypto.so.1.1 => /lib/libcrypto.so.1.1 (0x7fbc0ec00000)
        libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0x7fbc0efdf000)
Error relocating /usr/lib/libmsquic.so.2: EVP_seed_cbc: symbol not found
```

@wfurt could you help here?

Reply:
yes. good catch. 

https://git.alpinelinux.org/aports/tree/main/openssl/APKBUILD?h=3.15-stable#n110

it seems like the crypto on alpine is built with different options than Quic counter part. I can fix our test images. 

I would assume quic does not care so we can perhaps also update msquic defaults.